docs/tasks/LargeFileUploadTask.md

```diff
@@ -1,84 +1,142 @@
+	const fileObject = new StreamUpload(readStream, fileName, totalsize);
+    ```
+***Note*** - You can also have a customized `FileObject` implementation which contains the `sliceFile(range: Range)` function which implements the logic to split the file into ranges.
+
+**Initiate the LargefileUploadTask options with Progres Handler and Range Size**
+```typescript
+    const progress = (range?: Range, extraCallBackParams?: unknown) => {
+	};
+	const completed = (result?: UploadResult, extraCallBackParams?: unknown) => {
+	};
+	const failure = (error?: GraphClientError, extraCallBackParams?: unknown) => {
+	};
+	const progressCallBack: Progress = {
+		progress,
+		completed,
+		failure,
+		extraCallBackParams, // additional parameters to the callback
+	};
+	
+	const options: LargeFileUploadTaskOptions = {
+			rangeSize: 327680,
+			progressCallBack: progressCallBack,
+	    };
+```
+**Create a LargefileUploadTask object**
 
-## Uploading from NodeJS
+```typescript
+const uploadTask = new LargeFileUploadTask(client, fileObj, uploadSession, optionsWithProgress);
+const uploadResult:UploadResult = await uploadTask.upload();
+```
+`UploadResult` contains the `location`(received in the Outlook API header response) and the `responseBody` properties.
 
+## OneDriveLargeFileUploadTask.
+You can also use `OneDriveLargeFileUploadTask` which provides easier access to upload to OneDrive API.
+Example -
 ```typescript
-function uploadFile() {
-	fs.readFile("<PATH_OF_THE_FILE>", {}, function(err, file) {
-		if (err) {
-			throw err;
-		}
-		let fileName = "<NAME_OF_THE_FILE_WITH_EXTN>";
-		oneDriveLargeFileUpload(client, file, fileName)
-			.then((response) => {
-				console.log(response);
-				console.log("File Uploaded Successfully.!!");
-			})
-			.catch((error) => {
-				throw err;
-			});
-	});
-}
+		const progressCallBack: Progress = {
+			progress,
+			completed,
+			failure,
+			extraCallBackParams: true,
+		};
 
-async function oneDriveLargeFileUpload(client, file, fileName) {
-	try {
-		let options = {
+		const options: OneDriveLargeFileUploadOptions = {
 			path: "/Documents",
 			fileName,
 			rangeSize: 1024 * 1024,
+			progressCallBack,
 		};
-		const uploadTask = await OneDriveLargeFileUploadTask.create(client, file, options);
-		const response = await uploadTask.upload();
-		return response;
-	} catch (err) {
-		console.log(err);
-	}
+		const readStream = fs.createReadStream(`./fileName`);
+		const fileObject = new StreamUpload(readStream, fileName, totalsize);
+		or
+		const readFile = fs.readFileSync(`./fileName`);
+		const fileObject = new FileUpload(readStream, fileName, totalsize);
+
+		const uploadTask = await OneDriveLargeFileUploadTask.createTaskWithFileObject(client, fileObject, options);
+		const uploadResult:UploadResult = await uploadTask.upload();
 }
 ```
+**Note - The `OneDriveLargeFileUploadTask.createTaskWithFileObject` also handles the upload session creation.**
```

Comment:
notes are usually prefixed with `> Note: ` in the doc. It'd be nice to align. Also, there is a difference of style between here and above `*** Note: ***`.  Please align styles

Reply:
Done